src/devTools.js

```diff
@@ -187,12 +217,36 @@ class DevToolsEnhancer {
     }
   };
 
+  startWrapper = () => {
+    if (this.started || (this.socket && this.socket.getState() === this.socket.CONNECTING)) return;
+
+    if (!this.socketOptions.port) {
+      // no port provided - we should throw!
+      throw new Error('no port provided');
+    }
+    if (this.socketOptions.hostname) {
+      // hostname provided - don't look for it
+      this.start();
+    } else {
+      // obtain the hostname
+      this.urlPromise
+        .then((url) => {
+          this.socketOptions.hostname = url;
+          this.socketUrl = url;
+          this.start();
+        })
+        .catch((err) => {
+          console.log('Error obtaining socket url: ' + err.toString());
+        });
+    }
+  };
+
   start = () => {
     if (this.started || this.socket && this.socket.getState() === this.socket.CONNECTING) return;
 
     this.socket = socketCluster.connect(this.socketOptions);
 
-    this.socket.on('error', function (err) {
+    this.socket.on('error', (err) => {
```

Comment:
I think using `function` here causes an error when trying to refer to `this.errorCounts`. With lambda, we get the `this` from the outer scope.